utilities_common/module_base.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+#
+# module_base.py
+#
+# Utility helper for modules within SONiC
+
+from sonic_py_common import logger
+from utilities_common.util_base import UtilHelper
+
+SYSLOG_IDENTIFIER = "module_base"
+NOT_AVAILABLE = 'N/A'
+INVALID_MODULE_INDEX = -1
+
+# Load the UtilHelper class
+util = UtilHelper()
+
+# Global logger instance
+log = logger.Logger(SYSLOG_IDENTIFIER)
+
+
+class ModuleHelper:
+    def __init__(self):
+        # Global variable for platform chassis
+        self.platform_chassis = util.load_platform_chassis()
+        if not self.platform_chassis:
+            log.log_error("Failed to load platform chassis")
+
+    def reboot_module(self, module_name, reboot_type):
+        """
+        Reboot the specified module by invoking the platform API.
+
+        Args:
+            module_name (str): The name of the module to reboot.
+            reboot_type (str): The type of reboot requested for the module.
+
+        Returns:
+            bool: True if the reboot command was successfully sent, False otherwise.
+        """
+        module_index = util.try_get(self.platform_chassis.get_module_index, module_name, default=INVALID_MODULE_INDEX)
```

Comment:
The current util_base.py does not accept function arguments, may beed to be modified

Reply:
Thank you, I have added one more function try_get_args which takes arguments also as input.